packages/medusa/src/services/line-item.ts

```diff
@@ -184,108 +184,50 @@ class LineItemService extends TransactionBaseService {
 
   /**
    * Generate a single or multiple line item without persisting the data into the db
-   * @param variantIdOrData
-   * @param regionIdOrContext
-   * @param quantity
+   * @param variantData
    * @param context
    */
   async generate<
-    T = string | GenerateInputData | GenerateInputData[],
-    TResult = T extends string
-      ? LineItem
-      : T extends LineItem
-      ? LineItem
-      : LineItem[]
-  >(
-    variantIdOrData: T,
-    regionIdOrContext: T extends string ? string : GenerateLineItemContext,
-    quantity?: number,
-    context: GenerateLineItemContext = {}
-  ): Promise<TResult> {
+    T = GenerateInputData | GenerateInputData[],
```

Comment:
todo: This is a complete breaking changes of the existing method. Should we revert and just accept variantId or data accepting another type and then handle it? eventually we can create override declaration an mark all the previous api as deprecated to be able to remove them at somepoint.

cc @olivermrbl

Reply:
I've updated the service to not have breaking changes as we discussed